modin/pandas/test/test_io.py

```diff
@@ -2152,16 +2229,45 @@ def test_read_sql(self, tmp_path, make_sql_connection, read_sql_engine):
             con=sqlalchemy_connection,
         )
 
+        old_sql_engine = ReadSqlEngine.get()
         ReadSqlEngine.put(read_sql_engine)
         if ReadSqlEngine.get() == "Connectorx":
             modin_df = pd.read_sql(sql=query, con=conn)
         else:
             modin_df = pd.read_sql(
                 sql=query, con=ModinDatabaseConnection("sqlalchemy", conn)
             )
+        ReadSqlEngine.put(old_sql_engine)
         pandas_df = pandas.read_sql(sql=query, con=sqlalchemy_connection)
         df_equals(modin_df, pandas_df)
 
+    @pytest.mark.xfail(
+        condition="config.getoption('--simulate-cloud').lower() != 'off'",
+        reason="The reason of tests fail in `cloud` mode is unknown for now - issue #3264",
+    )
+    @pytest.mark.parametrize(
+        "dtype_backend", [lib.no_default, "numpy_nullable", "pyarrow"]
+    )
+    def test_read_sql_dtype_backend(self, tmp_path, make_sql_connection, dtype_backend):
+        filename = get_unique_filename(extension="db")
+
+        table = "test_read_sql_dtype_backend"
+        conn = make_sql_connection(tmp_path / filename, table)
+        query = f"select * from {table}"
+
+        def comparator(df1, df2):
+            df_equals(df1, df2)
+            df_equals(df1.dtypes, df2.dtypes)
```

Comment:
feels like it should be extracted to a common place, this comparator is written in a few places already; could go in some tests refactoring PR I suppose